### ISSUE TITLE
Grading table with collections

### ISSUE THREAD

**Issue by ParticuleAlpha** [cgnieder/xsim#85]
Hi!
I've updated the package this morning as I was working on building a new document I've noticed that the grading table does not work properly with collections since then.

Here is a minimal (non-)working example (that I got from issue #40 ):
```latex
\documentclass{article}
\usepackage{xsim}

\DeclareExerciseCollection{test}
\xsimsetup{print-solutions/headings=false}

\begin{filecontents*}{testproblems.tex}
\begin{exercise}[ID=one,points=3]
  Exercise one.
\end{exercise}
\begin{solution}
  Answer one.
\end{solution}

\begin{exercise}[ID=two,points=7]
  Exercise two.
\end{exercise}
\begin{solution}
  Answer two.
\end{solution}

\begin{exercise}[ID=three,points=1]
  Exercise three.
\end{exercise}
\begin{solution}
  Answer three.
\end{solution}
\end{filecontents*}

\begin{document}

\gradingtable

\collectexercises{test}
  \input{testproblems.tex}
\collectexercisesstop{test}

\printexercise{exercise}{one,three}

\end{document}
```

And here is the result I have:
![image](https://user-images.githubusercontent.com/80419737/110678743-ef238900-81d6-11eb-8848-d912d8515a2b.png)

It looks kind of like it does not "see" the exercises.

Can you help me figuring out what I'm doing wrong?

PS: when exercises are included directly in the document body, without using collections, it works perfectly fine.

**Comment by cgnieder** [comment]
I suggest to read the current manual carefully again. The long announced changed to the collection mechanism finally took place with version 0.20 and is documented. This version works:

```latex
\documentclass{article}
\usepackage{xsim}

\DeclareExerciseCollection{test}
\xsimsetup{
  print-solutions/headings=false ,
  collect % <<<<<
}

\begin{document}

\gradingtable

\begin{exercise}[ID=one,points=3]
  Exercise one.
\end{exercise}
\begin{solution}
  Answer one.
\end{solution}

\begin{exercise}[ID=two,points=7]
  Exercise two.
\end{exercise}
\begin{solution}
  Answer two.
\end{solution}

\begin{exercise}[ID=three,points=1]
  Exercise three.
\end{exercise}
\begin{solution}
  Answer three.
\end{solution}

\printexercise{exercise}{one,three}

\end{document}
```
![image](https://user-images.githubusercontent.com/2248538/111064177-deaf2f00-84b2-11eb-871f-6ed929d82f9c.png)

**Comment by ParticuleAlpha** [comment]
Thank you very much for your answer, I somehow managed to totally miss it in the manual even if I read it more than once, sorry about that!

I have a supplementary question about it. If you add the second exercise in the printed exercises ( using `\printexercise{exercise}{one,three,two}` in your MWE), it gives you a grading table with numbers ordered by `id` instead of by `counter` (as it should per the definition of functions in the grading table template) and exercise 2 is at the end of the table instead of being between exercise 1 and exercise 3.

![image](https://user-images.githubusercontent.com/80419737/111269496-02977f80-862f-11eb-8b98-7409123b865a.png)

I've tried to change the grading table template without success so far. Is there a build-in (or an easy) way to do so that I overlooked?
